packages/flutter/lib/src/cupertino/menu_anchor.dart

```diff
@@ -0,0 +1,3136 @@
+  });
+
+  /// An optional controller that allows opening and closing of the menu from
+  /// other widgets.
+  final MenuController? controller;
+
+  /// A callback invoked when the menu is opened while having an
+  /// [AnimationStatus] of [AnimationStatus.dismissed] or [AnimationStatus.reverse].
+  final VoidCallback? onOpen;
+
+  /// A callback invoked when the menu is closed while having an
+  /// [AnimationStatus] of [AnimationStatus.complete] or [AnimationStatus.forward].
+  final VoidCallback? onClose;
+
+  /// A callback that is invoked when the status of the menu changes.
+  ///
+  /// Unlike [onOpen] and [onClose], this callback is invoked for all
+  /// [AnimationStatus] changes.
+  final CupertinoMenuAnimationStatusChangedCallback? onAnimationStatusChange;
+
+  /// The point on the anchor surface that attaches to the menu.
+  ///
+  /// This value is ignored if the menu is opened with an explicit position.
+  final AlignmentGeometry? alignment;
+
+  /// The offset of the menu relative to the alignment origin determined by
+  /// [alignment] and the ambient [Directionality].
+  ///
+  /// Use this for adjustments of the menu placement.
+  ///
+  /// Increasing [Offset.dy] values of [alignmentOffset] move the menu position
+  /// down.
+  ///
+  /// If the [alignment] is an [AlignmentDirectional] AND the text direction is
+  /// [TextDirection.rtl], a larger [Offset.dx] component of [alignmentOffset]
+  /// moves the menu position to the left. Otherwise, a larger [Offset.dx] moves
+  /// the menu position to the right.
+  ///
+  /// This value is ignored if the menu is opened with an explicit position.
+  final ui.Offset? alignmentOffset;
+
+  /// The point on the menu surface that attaches to the anchor.
+  final AlignmentGeometry? menuAlignment;
+
+  /// The constraints to apply to the menu scrollable.
+  final BoxConstraints? constraints;
+
+  /// Whether the menu's cross axis should be constrained by the overlay.
+  ///
+  /// If true, when the menu is wider than the overlay, the menu width will
+  /// shrink to fit the overlay bounds.
+  ///
+  /// If false, the menu will grow to fit the size of its contents. If the menu
+  /// is wider than the overlay, it will be clipped to the overlay's bounds.
+  ///
+  /// Defaults to false.
+  final bool constrainCrossAxis;
+
+  /// Whether or not a tap event that closes the menu will be permitted to
+  /// continue on to the gesture arena.
+  ///
+  /// If false, then tapping outside of a menu when the menu is open will both
+  /// close the menu, and allow the tap to participate in the gesture arena. If
+  /// true, then it will only close the menu, and the tap event will be
+  /// consumed.
+  ///
+  /// Defaults to false.
+  final bool consumeOutsideTaps;
+
+  /// Whether or not swiping is enabled on the menu.
+  ///
+  /// When swiping is enabled, a [MultiDragGestureRecognizer] is added around
+  /// the menu button and menu items. The [MultiDragGestureRecognizer] allows
```

Comment:
I recommend we do not  use the name "menu button" since we have not defined it. Either define it somewhere else or describe it similar to this:
```suggestion
  /// the widget built by [builder] and menu items. The [MultiDragGestureRecognizer] allows
```